### PR TITLE
Support of collections

### PR DESCRIPTION
**Pull request by laurentj**
Support of the syntax to declare collections as described in psr-5, like `collectionobject<valuetype>` or `collectionobject<keytype,valuetype>`.

### CODE REVIEW

File: src/Types/Collection.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+     * Returns a rendered output of the Type as it would be used in a DocBlock.
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        if ($this->keyType === null) {
+            return $this->fqsen.'<'.$this->valueType . '>';
+        }
+        return $this->fqsen.'<'.$this->keyType . ',' . $this->valueType . '>';
```
Comment (jaapio) [inline src/Types/Collection.php]: please add a newline here, for readability

File: tests/unit/CollectionResolverTest.php [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+/**
+ * @coversDefaultClass phpDocumentor\Reflection\TypeResolver
+ */
+class CollectionResolverTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     *
+     * @covers ::__construct
+     * @covers ::resolve
+     * @covers ::<private>
```
Comment (jaapio) [inline tests/unit/CollectionResolverTest.php]: When you move this `::<private>` to the class level you don't have to repeat it every time.

File: tests/unit/CollectionResolverTest.php [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage string is not a collection
+     */
+    public function testBadCollectionClass()
+    {
+        $fixture = new TypeResolver();
+        $fixture->resolve('string<integer>', new Context(''));
+
+    }
+}
```
Comment (jaapio) [inline tests/unit/CollectionResolverTest.php]: Missing newline

File: tests/unit/CollectionResolverTest.php [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+     * @covers ::resolve
+     * @covers ::<private>
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage An array can have only integers or strings as keys
+     */
+    public function testBadArrayCollectionKey()
+    {
+        $fixture = new TypeResolver();
+        $fixture->resolve('array<object,string>', new Context(''));
+
```
Comment (jaapio) [inline tests/unit/CollectionResolverTest.php]: Please remove these newlines

File: tests/unit/CollectionResolverTest.php [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+     * @covers ::resolve
+     * @covers ::<private>
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage Unexpected collection operator "<", class name is missing
+     */
+    public function testMissingStartCollection()
+    {
+        $fixture = new TypeResolver();
+        $fixture->resolve('<string>', new Context(''));
+
```
Comment (jaapio) [inline tests/unit/CollectionResolverTest.php]: Please remove these newlines

File: tests/unit/CollectionResolverTest.php [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+     * @covers ::resolve
+     * @covers ::<private>
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage string is not a collection
+     */
+    public function testBadCollectionClass()
+    {
+        $fixture = new TypeResolver();
+        $fixture->resolve('string<integer>', new Context(''));
+
```
Comment (jaapio) [inline tests/unit/CollectionResolverTest.php]: Please remove these newlines

**Comment by laurentj** [comment]
Here are requested changes :)

**Comment by jaapio** [comment]
Thanks again for this great job!

**Comment by laurentj** [comment]
with pleasure :)

**Pull request merged**